gprofiler/profilers/java.py

```diff
@@ -259,11 +285,109 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": self.get_jvm_flags(process)}
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    @functools.lru_cache(maxsize=1024)
+    def get_jvm_flags(self, process: Process) -> Optional[List[tuple]]:
+        jvm_raw_flags = self.get_jvm_flags_raw(process)
+        if jvm_raw_flags is None:
+            return None
+
+        return [dataclasses.astuple(flag) for flag in jvm_raw_flags if self.filter_jvm_flags(flag)]
+
+    @staticmethod
+    def filter_jvm_flags(flag: JvmFlag) -> bool:
```

Comment:
```suggestion
    def filter_jvm_flag(flag: JvmFlag) -> bool:
```

It filters a single flag.